### ISSUE TITLE
Validating CLI arguments

### ISSUE THREAD

**Issue by edublancas** [ploomber/jupysql#531]
I noticed that passing random arguments to `%sql` doesn't raise any errors:

```
(jupysql)  eduardo@macbookair dev/jupysql (display *) » ipython
Python 3.10.11 (main, Apr 20 2023, 13:58:42) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.13.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: %load_ext sql

In [2]: %sql duckdb://

In [3]: %sql --stuff --more-stuff
Running query in 'duckdb://'

In [4]:
```

We should throw an UsageError instead

**Comment by edublancas** [comment]
https://github.com/ploomber/jupysql/blob/master/src/sql/magic.py

**Comment by tl3119** [comment]
To solve this issue, we need to focus on the arguments part on the `execute()` method in `magic.py`, since we don't want random arguments to be passed.  
```
@magic_arguments(allow_interspersed_args=False)
@handle_arguments
def handle_arguments(self, *args):
    if len(args) > 0:
        raise UsageError("Unrecognized argument(s): {}".format(args))

@line_magic("sql")
@cell_magic("sql")
# Existing code
```
In this way, we prevents unrecognized arguments from being treated as positional arguments and to handle the error we got.
Does this method works fine?
BTW, how do we test for this?

**Comment by edublancas** [comment]
yeah, this makes sense: splitting into two. the root problem is that both the line and cell lines are executing the same function. but the validation logic assumptions are different (see https://github.com/ploomber/jupysql/issues/550)

so go ahead with the approach you suggested and we'll see how it goes!

**Comment by tl3119** [comment]
Sure! I have submitted a PR request. Some tests failed, I am blocked by these tests. Do you have any suggestion for these error?

**Comment by tl3119** [comment]
i have some ideas now, i will try to fix that

**Comment by edublancas** [comment]
> How do we distinguish between unrecognized argument and SQL comment? 

we cannot do this in the line magic (`%sql` - single percentage sign), so we won't implement this there.

but we can do it in the cell magic (`%%sql` - double percentage sign) because the SQL content begins in the second line. so we can assume the first line only has arguments

**Comment by tl3119** [comment]
Sure, that makes sense. I have submitted a new pull request. I added a new test, and all the tests pass locally in my computer.
